.github/workflows/docgen.yml

```
@@ -0,0 +1,43 @@
+name: Nightly
+
+on:
+  schedule:
+      - cron:  '0 12 * * *'
+  workflow_dispatch:
+
+jobs:
+  docgen:
+    name: Generate Reference Docs
+    if: github.repository_owner == 'withastro'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code using Git
+        uses: actions/checkout@v3
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.2.1
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'pnpm'
+
+      - name: Install dependencies
+        run: pnpm install
```

```suggestion
      - name: Install Tools & Dependencies
        uses: ./.github/actions/install
```
We actually have a shared action that performs these steps already! It would be great to keep this in sync with our CI workflow.